src/app/components/challenge/challengeviewallsubmissions/challengeviewallsubmissions.component.ts

```diff
@@ -209,6 +214,17 @@ export class ChallengeviewallsubmissionsComponent implements OnInit, AfterViewIn
     this.apiService.getUrl(API_PATH).subscribe(
       data => {
         SELF.submissions = data['results'];
+        for (let i = 0; i < SELF.submissions.length; i++) {
+          (SELF.submissions[i].is_public) ?
```

Comment:
A better way could be this:
`SELF.submissions[i].submissionVisibilityIcon = (SELF.submissions[i].is_public) ? 'visibility' : 'visibility_off';`

Reply:
done